### PR TITLE
Support crossjoin in substrait.

### PR DESCRIPTION
**Pull request by my-vegetable-has-exploded**
## Which issue does this PR close?

ref: #8149

Closes #.

## Rationale for this change

part of #8149.

## What changes are included in this PR?

Translate [cross product](https://substrait.io/relations/logical_relations/#cross-product-operation)  to LogicalPlan::CrossJoin.

### CODE REVIEW

File: datafusion/substrait/src/logical_plan/producer.rs [hunk 1]
```diff
@@ -332,6 +333,18 @@ pub fn to_substrait_rel(
                 }))),
             }))
         }
+        LogicalPlan::CrossJoin(cross_join) => {
+            let left = to_substrait_rel(cross_join.left.as_ref(), ctx, extension_info)?;
```
Comment (alamb) [inline datafusion/substrait/src/logical_plan/producer.rs:337]: 👍 
I double checked there are no extra fields that are missed: https://github.com/apache/arrow-datafusion/blob/2e5ad7a3cb3b3f3e96f931b903eb8bb6639329ff/datafusion/expr/src/logical_plan/plan.rs#L2050

One way to potentially make this code more future proof if we add fields to `CrossJoin` in the future (aka have the compiler tell about places that may need updates) would be to use a direct restructuring

Like

```suggestion
            let CrossJoin { left, right, schema: _} = cross_join
            let left = to_substrait_rel(left.as_ref(), ctx, extension_info)?;
```
...

**Review by alamb — APPROVED** [review]
Thank you for the contribution @my-vegetable-has-exploded this looks great to me. I had a small suggestion to improve the code but I don't think it is necessary

File: datafusion/substrait/src/logical_plan/producer.rs (hunk 1 above)
Comment (my-vegetable-has-exploded) [reply datafusion/substrait/src/logical_plan/producer.rs]: good idea.

File: datafusion/substrait/src/logical_plan/consumer.rs [hunk 2]
```diff
@@ -434,6 +434,16 @@ pub async fn from_substrait_rel(
...
         }
+        Some(RelType::Cross(cross)) => {
+            let left: LogicalPlanBuilder = LogicalPlanBuilder::from(
+                from_substrait_rel(ctx, cross.left.as_ref().unwrap(), extensions).await?,
+            );
+            let right = LogicalPlanBuilder::from(
+                from_substrait_rel(ctx, cross.right.as_ref().unwrap(), extensions)
+                    .await?,
+            );
+            left.cross_join(right.build()?)?.build()
```
Comment (waynexia) [inline datafusion/substrait/src/logical_plan/consumer.rs:445]: `LogicalPlanBuilder` to right side hand plan is unnecessary -- You create a builder from a plan and freeze it to plan immediately.

**Review by waynexia — APPROVED** [review]
Rest LGTM, thanks @my-vegetable-has-exploded

File: datafusion/substrait/src/logical_plan/consumer.rs (hunk 2 above)
Comment (my-vegetable-has-exploded) [reply datafusion/substrait/src/logical_plan/consumer.rs]: get it! thanks @waynexia

**Comment by alamb** [comment]
🎉

**Pull request merged**